docs/src/components/ColorSwatch/ColorSwatch.component.tsx

```diff
@@ -0,0 +1,104 @@
+// REACT
+import * as React from 'react';
+// VENDOR
+import styled, { css } from 'styled-components';
+import classNames from 'classnames';
+import { Pill } from '../Pill'
+import { CopyToClipboard } from 'react-copy-to-clipboard';
+// THEME
+import { colors, sizes } from '../../../../src/theme';
+
+
+// INTERFACES
+interface ColorSwatchProps {
+  className?: string;
+  backgroundColor: string;
+  contrastOnBlack: number;
+  contrastOnF1F1F1: string;
+  contrastOnFAFAFA: string;
+  contrastOnWhite: number;
+  copyChoice: string;
+  name: string;
+  size: 'large' | 'small';
+  textColor: string;
+}
+
+interface ColorSwatchStylesGroup {
+  [key: string]: FlattenSimpleInterpolation;
+}
+
+
+// STYLES
+const ColorSwatchSizeStyles: ColorSwatchStylesGroup = {
+  large: css`
+    .ColorSwatchPillContainer {
+      padding-top: 200px
+    }
+  `,
+  small: css`
+    .ColorSwatchPillContainer {
+      padding-top: 50px
+    }
+  `,
+}
+
+const StyledColorSwatch = styled.div`
+  background: ${props => props.backgroundColor };
+  color: ${props => props.textColor };
+  padding: 24px 16px 16px 16px;
+
+  .ColorSwatchName {
+    margin: 0;
+    font-weight: 600;
+  }
+
+  .ColorSwatchBackgroundColor {
+    margin: 0;
+  }
+
+  ${({ size = 'large' }: ColorSwatchProps) => ColorSwatchSizeStyles[size]}
+`;
+
+
+// JSX MARKUP
+export const ColorSwatch = ({
+  className,
+  backgroundColor,
+  contrastOnBlack,
+  contrastOnF1F1F1,
+  contrastOnFAFAFA,
+  contrastOnWhite,
+  copyChoice,
+  name,
+  size,
+  textColor
+}: ColorSwatchProps): React.ReactElement<any> => {
+
+  let copyToClipboardText = backgroundColor;
+
+  if (copyChoice == 'variableName') {
+    copyToClipboardText = name;
+  }
+
+  return (
+
+    <CopyToClipboard text={copyToClipboardText}>
+      <StyledColorSwatch
+        className={classNames(className)}
+        backgroundColor={backgroundColor}
+        textColor={textColor}
+        size={size}
+      >
+       <p class="ColorSwatchName">{name}</p>
+       <p class="ColorSwatchBackgroundColor">{backgroundColor}</p>
+       <div class="ColorSwatchPillContainer">
+         <Pill value={contrastOnWhite} backgroundColor="#ffffff" textColor={backgroundColor}></Pill>
+         <Pill value={contrastOnFAFAFA} backgroundColor="#fafafa" textColor={backgroundColor}></Pill>
```

Comment:
Minor suggestion: I suspect the pills would look more pleasing if they were a uniform width. Currently the width is being determined by however many digits the value happens to be.

Reply:
I like it. I updated to a uniform width ✅